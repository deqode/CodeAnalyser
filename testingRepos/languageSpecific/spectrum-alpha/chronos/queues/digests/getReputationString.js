// @flow
import {
  getReputationChangeInTimeframe,
  getTotalReputation,
} from '../../models/reputationEvent';
import type { Timeframe } from 'chronos/types';

export const getReputationString = ({
  totalReputation,
  reputationGained,
  timeframe,
}: {
  timeframe: Timeframe,
  totalReputation: number,
  reputationGained: number,
}) => {
  const hasGainedReputation = reputationGained > 0;
  const isFirstReputation = totalReputation === reputationGained;
  const during = timeframe === 'weekly' ? 'last week' : 'yesterday';

  let reputationString;
  if (hasGainedReputation) {
    reputationString = `You gained ${reputationGained} reputation ${during}.`;
  } else {
    reputationString = `You didnâ€™t gain any reputation ${during}.`;
  }

  if (isFirstReputation) {
    reputationString += ` Reputation is an indicator of how active and constructive you are across all your communities. The more great conversations you start or join, the more reputation you will have.`;
  } else {
    reputationString += ` You have a total of ${totalReputation} reputation across all of your communities${
      hasGainedReputation ? ' - well done!' : '.'
    }`;
  }

  return reputationString;
};

export default async (userId: string, timeframe: Timeframe) => {
  const [reputationGained, totalReputation] = await Promise.all([
    getReputationChangeInTimeframe(userId, timeframe),
    getTotalReputation(userId),
  ]);

  return getReputationString({
    totalReputation: totalReputation.toLocaleString().toString(),
    reputationGained: reputationGained.toLocaleString().toString(),
    timeframe,
  });
};
