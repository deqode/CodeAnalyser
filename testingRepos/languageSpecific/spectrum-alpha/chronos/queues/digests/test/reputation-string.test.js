// @flow
import { getReputationString } from '../getReputationString';

describe('first-time users', () => {
  it('should get an explanation what reputation is', () => {
    expect(
      getReputationString({
        timeframe: 'weekly',
        reputationGained: 10,
        totalReputation: 10,
      })
    ).toMatchSnapshot();
  });

  it("should be told when they haven't gained any reputation", () => {
    expect(
      getReputationString({
        timeframe: 'weekly',
        reputationGained: 0,
        totalReputation: 0,
      })
    ).toMatchSnapshot();
  });
});

describe('existing users', () => {
  it('should handle no reputation gain', () => {
    expect(
      getReputationString({
        timeframe: 'weekly',
        reputationGained: 0,
        totalReputation: 10,
      })
    ).toMatchSnapshot();
  });

  it('should handle reputation gain', () => {
    expect(
      getReputationString({
        timeframe: 'weekly',
        reputationGained: 10,
        totalReputation: 20,
      })
    ).toMatchSnapshot();
  });
});

it('should handle a daily timeframe', () => {
  expect(
    getReputationString({
      timeframe: 'daily',
      reputationGained: 10,
      totalReputation: 20,
    })
  ).toMatchSnapshot();

  expect(
    getReputationString({
      timeframe: 'daily',
      reputationGained: 0,
      totalReputation: 0,
    })
  ).toMatchSnapshot();
});
