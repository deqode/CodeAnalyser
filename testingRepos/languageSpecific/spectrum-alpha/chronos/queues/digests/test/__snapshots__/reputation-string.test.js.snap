// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`existing users should handle no reputation gain 1`] = `"You didn’t gain any reputation last week. You have a total of 10 reputation across all of your communities."`;

exports[`existing users should handle reputation gain 1`] = `"You gained 10 reputation last week. You have a total of 20 reputation across all of your communities - well done!"`;

exports[`first-time users should be told when they haven't gained any reputation 1`] = `"You didn’t gain any reputation last week. Reputation is an indicator of how active and constructive you are across all your communities. The more great conversations you start or join, the more reputation you will have."`;

exports[`first-time users should get an explanation what reputation is 1`] = `"You gained 10 reputation last week. Reputation is an indicator of how active and constructive you are across all your communities. The more great conversations you start or join, the more reputation you will have."`;

exports[`should handle a daily timeframe 1`] = `"You gained 10 reputation yesterday. You have a total of 20 reputation across all of your communities - well done!"`;

exports[`should handle a daily timeframe 2`] = `"You didn’t gain any reputation yesterday. Reputation is an indicator of how active and constructive you are across all your communities. The more great conversations you start or join, the more reputation you will have."`;
